webapp/channels/src/components/channel_invite_modal/team_warning_banner/team_warning_banner.tsx

```diff
@@ -0,0 +1,239 @@
+        const firstName = guests[0].username;
+        const lastName = guests[guests.length - 1].username;
+        if (guests.length > 10) {
+            return (
+                formatMessage(
+                    {
+                        id: t('channel_invite.invite_team_members.guests.messageOverflow'),
+                        defaultMessage: '{firstUser} and {others} are guest users and need to first be invited to the team before you can add them to the channel. Once they\'ve joined the team, you can add them to this channel.',
+                    },
+                    {
+                        firstUser: (
+                            <Markdown
+                                message={`@${firstName}`}
+                                options={{
+                                    atMentions: true,
+                                    mentionKeys,
+                                    mentionHighlight: false,
+                                    singleline: true,
+                                }}
+                            />
+                        ),
+                        others: (
+                            <SimpleTooltip
+                                id={'usernames-overflow'}
+                                content={commaSeparatedUsernames.replace(`@${firstName}, `, '')}
+                            >
+                                <span
+                                    className='add-others-link'
+                                >
+                                    <FormattedMessage
+                                        id='channel_invite.invite_team_members.messageOthers'
+                                        defaultMessage='{count} others'
+                                        values={{
+                                            count: guests.length - 1,
+                                        }}
+                                    />
+                                </span>
+                            </SimpleTooltip>
+                        ),
+                    },
+                )
+            );
+        }
+
+        const message: string = formatMessage(
+            {
+                id: t('channel_invite.invite_team_members.guests.message'),
+                defaultMessage: '{count, plural, =1 {{firstUser} is a guest user and needs} other {{users} and {lastUser} are guest users and need}} to first be invited to the team before you can add them to the channel. Once they\'ve joined the team, you can add them to this channel.',
+            },
+            {
+                count: guests.length,
+                users: commaSeparatedUsernames.replace(`, @${lastName}`, ''),
+                firstUser: `@${firstName}`,
+                lastUser: `@${lastName}`,
+                team: team.display_name,
+            },
+        );
+
+        return (
+            <Markdown
+                message={message}
```

Comment:
Chris previously mentioned:
> I don't think we are supposed to be putting translations though markdown formatting.

I'm not actually sure what the answer is here. @M-ZubairAhmed Do you have an answer?

Reply:
If I understand this code right, you're wanting this so that the at mentions are highlighted and have the profile popover, is that correct?

If so, instead of translating the message and then passing it into the Markdown component, I think you could flip that around by taking the list of users, rendering an `AtMention` component for each of them, passing them to a React Intl `FormattedList` to get the commas added between them, and then passing that to a `FormattedMessage`. Something like
```jsx
return (
     <FormattedMessage
          defaultMessage='{users} and {lastUser} etc...'
          values={{
              users: <FormattedList type='unit' style='short' value={[
                  <AtMention mentionName='user1'/>,
                  <AtMention mentionName='user2'/>,
              ]}/>,
              lastUser: <AtMention mentionName='user3'/>,
          }}
     />
);
```